lib/gridstore/src/bitmask/mod.rs

```diff
@@ -360,15 +360,39 @@ impl Bitmask {
         block_offset: BlockOffset,
         num_blocks: u32,
         used: bool,
+    ) {
+        let relative_range = block_offset as usize..(block_offset as usize + num_blocks as usize);
+        self.mark_blocks_batch(page_id, std::iter::once(relative_range), used);
+    }
+
+    /// Marks blocks sharing the same page in batch. First updates all ranges in the bitmask, then updates the region gaps a single time.
+    ///
+    /// # Arguments
+    /// * `page_id` - The ID of the page to mark blocks on.
+    /// * `block_ranges` - An iterator over the ranges of blocks to mark, relative to the page start.
+    /// * `used` - Whether the blocks should be marked as used or free.
+    pub(crate) fn mark_blocks_batch(
+        &mut self,
+        page_id: PageId,
+        relative_block_ranges: impl Iterator<Item = Range<usize>>,
```

Comment:
Nit: I vote to rename this to local:

```suggestion
        local_block_ranges: impl Iterator<Item = Range<usize>>,
```

Relative implies to me that each iterator item is relative to each other. While local implies to me the ranges are local within a page.